core/app/jobs/spree/promotions/duplicate_promotion_job.rb

```diff
@@ -0,0 +1,49 @@
+module Spree
+  module Promotions
+    class DuplicatePromotionJob < Spree::BaseJob
+      def perform(template_promotion_id:, batch_id:, options: {}, code: nil)
+        promotion = find_instance(template_promotion_id, model: Spree::Promotion)
+        code = code || applicable_code(batch_id, options)
+
+        Spree::PromotionHandler::PromotionBatchDuplicator.new(promotion, batch_id, code: code).duplicate
+      rescue CodeGenerator::MutuallyExclusiveInputsError, ActiveRecord::RecordNotUnique => e
+        Rails.logger.error(e.message)
```

Comment:
It does handle it - the job completes and isn't being rerun as it'd produce the same error.

```
2023-12-22T09:55:37.377Z pid=12726 tid=1a0e class=Spree::Promotions::DuplicatePromotionJob jid=bcd0cd6bce535a8df0b5ff26 INFO: start
2023-12-22T09:55:37.384Z pid=12726 tid=1a0e class=Spree::Promotions::DuplicatePromotionJob jid=bcd0cd6bce535a8df0b5ff26 INFO: Performing Spree::Promotions::DuplicatePromotionJob (Job ID: 711480c5-caed-4020-a5fc-4778a45e796a) from Sidekiq(default) enqueued at 2023-12-22T09:55:37.376591000Z with arguments: {:template_promotion_id=>4, :batch_id=>"7", :options=>{:affix=>:prefix, :content=>"foo", :deny_list=>["foo"]}}
2023-12-22T09:55:37.395Z pid=12726 tid=1a0e class=Spree::Promotions::DuplicatePromotionJob jid=bcd0cd6bce535a8df0b5ff26 ERROR: Invalid inputs. Forbidden phrases cannot contain the affix.
2023-12-22T09:55:37.395Z pid=12726 tid=1a0e class=Spree::Promotions::DuplicatePromotionJob jid=bcd0cd6bce535a8df0b5ff26 INFO: Performed Spree::Promotions::DuplicatePromotionJob (Job ID: 711480c5-caed-4020-a5fc-4778a45e796a) from Sidekiq(default) in 11.45ms
2023-12-22T09:55:37.395Z pid=12726 tid=1a0e class=Spree::Promotions::DuplicatePromotionJob jid=bcd0cd6bce535a8df0b5ff26 elapsed=0.018 INFO: done

Reply:
Won't this mark the job as succeeded? Maybe automatic retries should be disabled for this job for these exceptions and they should be propagated normally?

Reply:
What do you mean by propagating normally?
Do you mean not rescuing the errors and adding `sidekiq_options retry: false`?

Reply:
Just for reference, since some conversations happened off-PR, I think propagating the error normally for the sake of a consistent API (task can be treated as failed by external tooling) would be preferable in my opinion